memory/src/main/scala/filodb.memory/format/BinaryVector.scala

```diff
@@ -119,7 +125,14 @@ trait VectorDataReader extends AvailableReader {
    * @param vector the BinaryVectorPtr native address of the BinaryVector
    * @param startElement the starting element # in the vector, by default 0 (the first one)
    */
-  def iterate(vector: BinaryVectorPtr, startElement: Int = 0): TypedIterator
+  def iterate(acc: MemoryAccessor, vector: BinaryVectorPtr, startElement: Int = 0): TypedIterator
+
+  def toBytes(acc: MemoryAccessor, vector: BinaryVectorPtr): Array[Byte] = {
+    val numByts = numBytes(acc, vector) + 4
+    val bytes = new Array[Byte](numByts)
```

Comment:
This method is used only for testing. One could delegate to accessor, but I think this is more of a capability that needs to be handled by caller since the length of the result array is also in question. Dont want to over-optimize here since it is used only in tests right now.

I will document.